google-cloud-firestore/src/main/java/com/google/cloud/firestore/TransactionRunner.java

```diff
@@ -181,6 +205,7 @@ private ApiFuture<T> restartTransactionCallback(Throwable throwable) {
     ApiException apiException = (ApiException) throwable;
     if (transaction.hasTransactionId() && isRetryableTransactionError(apiException)) {
       if (attemptsRemaining > 0) {
+        getTraceUtil().currentSpan().addEvent("Initiate transaction retry");
```

Comment:
just a thought - maybe add attempts remaining count to the event string?

Reply:
Good idea. Although note that this information will also surface as part of the span attributes for the next attempt's span (line 96 above). But I think it's good surface it here as well. Done.